Quadriga/src/main/java/edu/asu/spring/quadriga/web/config/SecurityContext.java

```diff
@@ -0,0 +1,171 @@
+        private SavedRequestAwareAuthenticationSuccessHandler authSuccessHandler;
+
+        @Override
+        public void configure(WebSecurity web) throws Exception {
+         // Spring Security ignores request to static resources such  as CSS or JS files.
+            web.ignoring().antMatchers("/static/**");
+        }
+
+        @Override
+        protected void configure(HttpSecurity http) throws Exception {
+            HeadersConfigurer<HttpSecurity> config = http.antMatcher("**").headers().frameOptions().sameOrigin();
+            config.and().exceptionHandling().accessDeniedHandler(accessDeniedHandler())
+                    // Configures url based authorization
+                    .and().formLogin().loginPage("/login").permitAll().defaultSuccessUrl("/auth/welcome")
+                    .successHandler(authSuccessHandler).failureUrl("/loginfailed").and().authorizeRequests()
+                    // Anyone can access the urls
+                    .antMatchers("/signin/**", "/connect/**", "/signup/**", "/user/register/**",
+                            "/resources/**", "/register*", "/login*", "/logout*", "/createaccount*", "/sites/**",
+                            "/public/**", "/search/**", "/home/**", "/sociallogin*", "/forbidden")
+                    .permitAll()
+                    // The rest of the our application is protected.
+                    .antMatchers("/users/**", "/admin/**").hasRole("QUADRIGA_USER_ADMIN")
+                    .antMatchers("/", "/auth/welcome**", "/auth/home**", "/auth/about**", "/auth/profile/**",
+                            "/rest/**")
+                    .hasAnyRole("QUADRIGA_USER_ADMIN", "QUADRIGA_USER_STANDARD", "QUADRIGA_USER_COLLABORATOR")
+                    .antMatchers("/auth/workbench/**", "/auth/rest/**", "/auth/conceptcollections/**",
+                            "/auth/conceptdetails/**", "/auth/transformation/**", "/auth/searchitems/**",
+                            "/auth/dictionaries/**", "/auth/editing/**", "/auth/networks/**")
+                    .hasAnyRole("QUADRIGA_USER_STANDARD", "QUADRIGA_USER_COLLABORATOR")
+                    .antMatchers("/users/**", "/admin/**", "/auth/users/**", "/checks/**")
+                    .hasRole("QUADRIGA_USER_ADMIN").anyRequest()
+                    .hasAnyRole("QUADRIGA_USER_ADMIN", "QUADRIGA_USER_STANDARD", "QUADRIGA_USER_COLLABORATOR")
+                    .antMatchers("/**").denyAll()
+                    // Adds the SocialAuthenticationFilter to Spring Security's
+                    // filter chain.
+                    .and().apply(new SpringSocialConfigurer());
+
+        }
+
+        @Autowired
+        @Override
+        protected void configure(AuthenticationManagerBuilder auth) throws Exception {
+            auth.authenticationProvider(daoAuthProvider());
+            auth.eraseCredentials(false);
+            auth.inMemoryAuthentication().withUser("admin").password("admin").roles("QUADRIGA_USER_ADMIN");
```

Comment:
Noooooo!!!! this needs to be configurable. At least the admin password needs to come from the pom, or any installation of Quadriga will always have an admin account admin/admin. SECURITY (to channel Abhijith...) ;)

Reply:
Is this admin the same as quadriga admin? I have created a new property called admin_username=${admin.username} and admin_password=${admin.properties} in the settings.properties file. Is that the correct approach?